low-level/solution/lr2/server/manifest.py

```diff
@@ -0,0 +1,348 @@
+#!/bin/python3
+
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com
+
+import errno
+import time
+import json
+import os
+
+from cortx.utils.process import SimpleProcess
+from cortx.utils.kv_store import KvStoreFactory
+from cortx.utils.discovery.error import ResourceMapError
+from framework.utils.conf_utils import GLOBAL_CONF, Conf, NODE_TYPE_KEY
+from framework.base.sspl_constants import (MANIFEST_SVC_NAME, LSHW_FILE,
+    MANIFEST_OUTPUT_FILE)
+from framework.utils.service_logging import CustomLog, logger
+from framework.utils.utility import Utility
+from framework.platforms.server.software import Service
+from framework.platforms.server.platform import Platform
+from server_resource_map import ServerResourceMap
+
+
+class ServerManifest():
+    """
+    ServerManifest class provides resource map and related information
+    like health.
+    """
+
+    name = "server_manifest"
+
+    def __init__(self):
+        """Initialize server manifest."""
+        super().__init__()
+        self.log = CustomLog(MANIFEST_SVC_NAME)
+        server_type = Conf.get(GLOBAL_CONF, NODE_TYPE_KEY)
+        # import pdb; pdb.set_trace()
+        Platform.validate_server_type_support(self.log, ResourceMapError, server_type)
+        self.field_mapping = {
+            'id': 'uid',
+            'class': 'type',
+            'description': 'description',
+            'product': 'product',
+            'serial': 'serial_number',
+            'vendor': 'manufacturer',
+            'part_number': 'part_number',
```

Comment:
most of the keys are different from lshw and actual manifest defined key. We kept all the fields which are required for manifest and even both key is the same. because we are trying to make generic logic will work in all cases. otherwise, we have to write unnecessary checks/filters in logic for particular lshw keys.
Check line 190 to 192 https://github.com/Seagate/cortx-monitor/blob/3cf7d1debdb8b6b8e9a39c59d265ae538339ccf8/low-level/solution/lr2/server/manifest.py#L190